simde/x86/gfni.h

```diff
@@ -0,0 +1,765 @@
+
+      SIMDE__VECTORIZE
+      for (size_t i = 0 ; i < (sizeof(r_.u8) / sizeof(r_.u8[0])) ; i++) {
+	q = HEDLEY_STATIC_CAST(uint64_t, x_.u8[i]) * ones;
+	q &= A_.u64[i / 8];
+	q ^= q >> 4;
+	q ^= q >> 2;
+	q ^= q >> 1;
+	q &= ones;
+	q *= 255;
+	q &= mask;
+	q |= q >> 32;
+	q |= q >> 16;
+	q |= q >> 8;
+	q ^= b;
+	r_.u8[i] = HEDLEY_STATIC_CAST(uint8_t, q);
+      }
+
+      return simde__m128i_from_private(r_);
+    #endif
+  }
+  #if defined(SIMDE_GFNI_ENABLE_NATIVE_ALIASES)
+    #define _mm_gf2p8affine_epi64_epi8(x, A, b) simde_mm_gf2p8affine_epi64_epi8(x, A, b)
+  #endif
+#endif
+
+#if defined(SIMDE_GFNI_NATIVE) && defined(SIMDE_AVX512VL_NATIVE)
+  #define simde_mm256_gf2p8affine_epi64_epi8(x, A, b) _mm256_gf2p8affine_epi64_epi8(x, A, b)
+#else
+  SIMDE__FUNCTION_ATTRIBUTES
+  simde__m256i
+  simde_mm256_gf2p8affine_epi64_epi8 (simde__m256i x, simde__m256i A, int b) {
+    simde__m256i_private
+      r_,
+      x_ = simde__m256i_to_private(x),
+      A_ = simde__m256i_to_private(A);
+
+    SIMDE__VECTORIZE
+    for (size_t i = 0 ; i < (sizeof(r_.m128i) / sizeof(r_.m128i[0])) ; i++) {
+      r_.m128i[i] = simde_mm_gf2p8affine_epi64_epi8(x_.m128i[i], A_.m128i[i], b);
+    }
+
+    return simde__m256i_from_private(r_);
+  }
+  #if defined(SIMDE_GFNI_ENABLE_NATIVE_ALIASES)
+    #define _mm256_gf2p8affine_epi64_epi8(x, A, b) simde_mm256_gf2p8affine_epi64_epi8(x, A, b)
+  #endif
+#endif
+
+#if defined(SIMDE_GFNI_NATIVE) && defined(SIMDE_AVX512F_NATIVE)
+  #define simde_mm512_gf2p8affine_epi64_epi8(x, A, b) _mm512_gf2p8affine_epi64_epi8(x, A, b)
+#else
+  SIMDE__FUNCTION_ATTRIBUTES
+  simde__m512i
+  simde_mm512_gf2p8affine_epi64_epi8 (simde__m512i x, simde__m512i A, int b) {
+    simde__m512i_private
+      r_,
+      x_ = simde__m512i_to_private(x),
+      A_ = simde__m512i_to_private(A);
+
+    SIMDE__VECTORIZE
+    for (size_t i = 0 ; i < (sizeof(r_.m128i) / sizeof(r_.m128i[0])) ; i++) {
+      r_.m128i[i] = simde_mm_gf2p8affine_epi64_epi8(x_.m128i[i], A_.m128i[i], b);
+    }
+
+    return simde__m512i_from_private(r_);
+  }
+  #if defined(SIMDE_GFNI_ENABLE_NATIVE_ALIASES)
+    #define _mm512_gf2p8affine_epi64_epi8(x, A, b) simde_mm512_gf2p8affine_epi64_epi8(x, A, b)
+  #endif
+#endif
+
+#if defined(SIMDE_GFNI_NATIVE) && defined(SIMDE_AVX512VL_NATIVE)
+  #define simde_mm_mask_gf2p8affine_epi64_epi8(src, k, x, A, b) _mm_mask_gf2p8affine_epi64_epi8(src, k, x, A, b)
+#else
+  SIMDE__FUNCTION_ATTRIBUTES
+  simde__m128i
+  simde_mm_mask_gf2p8affine_epi64_epi8 (simde__m128i src, simde__mmask16 k, simde__m128i x, simde__m128i A, int b) {
+    simde__m128i_private
+      src_ = simde__m128i_to_private(src);
+
+    simde__m128i r = simde_mm_gf2p8affine_epi64_epi8(x, A, b);
+    simde__m128i_private r_ = simde__m128i_to_private(r);
+
+    SIMDE__VECTORIZE
+    for (size_t i = 0 ; i < (sizeof(r_.u8) / sizeof(r_.u8[0])) ; i++) {
+      r_.u8[i] = ((k >> i) & 1) ? r_.u8[i] : src_.u8[i];
+    }
+
+    return simde__m128i_from_private(r_);
+  }
```

Comment:
AFAICT this could/should just be `return simde_mm_mask_mov_epi8(src, k, simde_mm_gf2p8affine_epi64_epi8(x, A, b));`  Same idea for the other mask functions.

Reply:
Yes, a good idea.
I think it should also enable me to clean up some of the b as a compile-time constant problems.
I originally wrote everything thinking b was a normal variable.
I did last-minute corrections but didn't cover all internal cases.